packages/react/react/src/hooks/useTimeout.ts

```diff
@@ -1,24 +1,14 @@
-import { useEffect, useRef } from 'react';
+import { noop } from '@toss/utils';
+import { useEffect } from 'react';
+import { usePreservedCallback } from './usePreservedCallback';
 
 /** @tossdocs-ignore */
-export function useTimeout(callback: () => void, delay = 0) {
-  const savedCallback = useRef<() => void>();
+export function useTimeout(callback = noop, delay = 0) {
```

Comment:
Other hooks like `useImpressionRef` already use `noop` as the default value for the callback function.

```tsx
export function useImpressionRef<Element extends HTMLElement>({
  onImpressionStart: _onImpressionStart = noop, // (*)
  onImpressionEnd: _onImpressionEnd = noop, // (*)
  // ...
}: Options)
```

Also, if we set the default value like this, we don't need to use conditional statements to determine whether the callback function exists or not.

I think it would make the code more readable.